### DISCUSSION TITLE
Nemo: Unifying sidebars

### DISCUSSION THREAD

**Discussion by jwueller** [linuxmint/discussions#478] [Pull requests]
It seems to me that there is quite a bit of overlap, but also exclusivity in the functionality of the tree and "Places" views of the sidebar. I propose merging them:

1. Convert currently hard-coded "Places" into bookmarks.
2. Convert currently hard-coded tree view roots into bookmarks (`/`, `~`, etc.)
3. Show bookmarks as individual root nodes in the tree view sidebar, so they can be expanded like expected.
4. Show dynamic devices as dynamic root nodes too, just like they currently do in "Places".
5. Remove the dedicated "Places" sidebar view.

Using the existing bookmarks feature is nice, because it reuses existing editing, sorting and storage capabilities in a backwards-compatible way. In addition, this would address a lot of open customization feature requests, while retaining reasonable defaults currently shown by "Places".

It seems to me that this would be the best of both worlds, because it affords a huge amount of flexibility while reducing complexity quite a bit, both in the user interface and maintenance. It's even nicely backwards compatible.

I'd also retain the nice themed icons used in "Places" based on where the bookmark is pointing, and potentially also port the disk usage indicators to the tree view. It could potentially even be used to show relative sizes of folders amongst each other further down the tree, `ncdu`-style. But that might be further down the line.

I already have a proof of concept working locally, but I'd hold off on fleshing it out until there is a consensus that this kind of PR would be accepted in the first place.

**Edit:** Adding the screenshot of my proof of concept in the main post too, for reference:
![image](https://github.com/linuxmint/discussions/assets/388308/cfdd0b74-8d77-4bdc-9e7f-20368138ddcd)

Please let me know what you think!

**Comment by Jopp-gh** [comment]
Subscribed. Nemo needs some update, imo. 

You may be interested in my proposal too:
https://github.com/orgs/linuxmint/discussions/430

**Comment by mtwebster** [comment]
I feel like this would increase complexity, not decrease.  The places sidebar is *all* I (and I'm guessing most other users) ever use. I've got the main list view already for drilling down thru subdirectories, I've considered getting rid of the tree sidebar because of its complexity and redundancy - I'm not interested in combining the two.

All levels of users need to be accounted for - many don't know what root or ~ or anything like that means - but they usually understand bookmarks and places, and want their removable drives to show up.

**Reply by jwueller** [reply]
I feel like I might not have communicated my proposal properly.

I'm not proposing showing `~` or `/` to the user. The labels and icons from "Places" would remain, as well as removable drives showing up. But the "Places" sidebar has a major limitation that the tree view doesn't: It doesn't support expansion. Using the tree view as a base provides that for free.

Could you elaborate on how this would increase complexity for users? You only get the additional arrow for those entries, which the "Places" view also uses already, albeit in a very limiting way.

Here is a screenshot of my 30 minute hack to try unifying them, showing the _tree view_, but including a handful of "Places" entries I moved over. Note how they also use the correct icons for those, because it's just inferred from the URI.

![image](https://github.com/linuxmint/discussions/assets/388308/cfdd0b74-8d77-4bdc-9e7f-20368138ddcd)

Note that this doesn't deal with devices or proper sorting yet, since it's just a quick demo.

**Comment by JosephMcc** [comment]
I'm with @mtwebster  on this one. I only use the places view. It's simple, to the point, and does what I want. This is just more complicated and messy.

**Comment by carsten88** [comment]
I use the folder-tree all the time. Browsing in compact view, for less scrolling.
If you use only paces, then you need to be in detailed view to navigate.
But tomato-tomato, everyone has their favorite way of navigating. 
I think Nemo should keep supporting different ways the user uses the software.

**Reply by cplittlfield** [reply]
I use the Places view to get to certain key locations in my system, but I really dislike having to take my hands off the keyboard.  Using a pointer to do anything is usually far less efficient.

I'd probably use the side panel more if there were a way to navigate to it from the main panel with keystrokes.  I don't use the Tree view at all, because navigating through it requires using my pen or a mouse, whereas in the main panel, I can quickly navigate with keystrokes.  Of course, the Terminal is the perfect tool for a PC-DOS baby like me.
